lib/monitor.sh

```diff
@@ -0,0 +1,47 @@
+
+monitor_memory_usage() {
+  local output_file=$1
+  local command="${@:2}"
+
+  # Run the command in the background
+  exec $command &
+
+  # save the PID of the running command
+  pid=$!
+
```

Comment:
I think you might want a `trap "kill -9 $pid" EXIT` here to prevent a build from hanging if the process manager only sends a signal to this process.

Reply:
If you do that, remember to `wait $pid` twice further down, because the trap will unblock the `wait`. Might be better to trap on `TERM` in this case (and `kill $pid` so it can shut down properly). Welcome to bash process management hell; AMA 🙃